feature/experimental/gribi/otg_tests/programming_with_reload/programming_with_reload_test.go

```diff
@@ -0,0 +1,537 @@
+	dc := gnmi.OC()
+	i := &oc.Interface{}
+	i.Enabled = ygot.Bool(state)
+	i.Type = ethernetCsmacd
+	i.Name = ygot.String(p.Name())
+	gnmi.Update(t, args.dut, dc.Interface(p.Name()).Config(), i)
+}
+
+// addStaticRoute configures static route needed for decap path
+func addStaticRoute(t *testing.T, dut *ondatra.DUTDevice, ip string) {
+	d := gnmi.OC()
+	s := &oc.Root{}
+	static := s.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut)).GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, deviations.StaticProtocolName(dut))
+	ipv4Nh := static.GetOrCreateStatic(innerDstIP1 + "/" + mask).GetOrCreateNextHop("0")
+	ipv4Nh.NextHop, _ = ipv4Nh.To_NetworkInstance_Protocol_Static_NextHop_NextHop_Union(ip)
+	gnmi.Update(t, dut, d.NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, deviations.StaticProtocolName(dut)).Config(), static)
+}
+
+// Reload chassis
+func (args *testArgs) reloadChassis(t *testing.T) {
+
+	gnoiClient, err := args.dut.RawAPIs().BindingDUT().DialGNOI(context.Background())
+	if err != nil {
+		t.Fatalf("Error dialing gNOI: %v", err)
+	}
+	bootTimeBeforeReboot := gnmi.Get(t, args.dut, gnmi.OC().System().BootTime().State())
+	t.Logf("DUT boot time before reboot: %v", bootTimeBeforeReboot)
+
+	rebootRequest := &spb.RebootRequest{
+		Method:  spb.RebootMethod_COLD,
+		Delay:   rebootDelay * oneSecondInNanoSecond,
+		Message: "Reboot chassis with delay",
+		Force:   true,
+	}
+
+	t.Logf("Send reboot request: %v", rebootRequest)
+	rebootResponse, err := gnoiClient.System().Reboot(context.Background(), rebootRequest)
+	t.Logf("Got reboot response: %v, err: %v", rebootResponse, err)
+	if err != nil {
+		t.Fatalf("Failed to reboot chassis with unexpected err: %v", err)
+	}
+
+	if rebootRequest.GetDelay() > 1 {
+		t.Logf("Validating DUT remains reachable for at least %d seconds", rebootDelay)
+		prevTime, err := time.Parse(time.RFC3339, gnmi.Get(t, args.dut, gnmi.OC().System().CurrentDatetime().State()))
+		if err != nil {
+			t.Fatalf("Failed parsing current-datetime: %s", err)
+		}
+		start := time.Now()
+		for {
+			time.Sleep(10 * time.Second)
+			t.Logf("Time elapsed %.2f seconds since reboot was requested.", time.Since(start).Seconds())
+			if uint64(time.Since(start).Seconds()) > rebootDelay {
+				t.Logf("Time elapsed %.2f seconds > %d reboot delay", time.Since(start).Seconds(), rebootDelay)
+				break
+			}
+			latestTime, err := time.Parse(time.RFC3339, gnmi.Get(t, args.dut, gnmi.OC().System().CurrentDatetime().State()))
+			if err != nil {
+				t.Fatalf("Failed parsing current-datetime: %s", err)
+			}
+			if latestTime.Before(prevTime) || latestTime.Equal(prevTime) {
+				t.Errorf("Get latest system time: got %v, want newer time than %v", latestTime, prevTime)
+			}
+			prevTime = latestTime
+		}
+	}
+
+	startReboot := time.Now()
+	t.Logf("Wait for DUT to boot up by polling the telemetry output.")
+	for {
```

Comment:
gnmi.Get is used as a trigger to declare that the node is done with the boot process (rebooted successfully) in gNOI reboot tests
https://github.com/openconfig/featureprofiles/blob/332b7d49a98712cadda876a0096ed3a8cbe109be/feature/gnoi/system/tests/complete_chassis_reboot/complete_chassis_reboot_test.go#L155-L170, and seemingly was just taken from there 

But here, we should try to call gRIBI servers immediately to validate UNAVAILABLE grpc status (otherwise, chances are that the test may miss a period where gRIBI is unable to process the request but does not return the status as expected).

Reply:
You're right, gNMI.GET call confirms device rebooted successfully. Inorder to address the comment, I am changing the implementation to following,

- reload router
- start gribi client connection in 1 min (1 minute timer is to make sure device begins the reload), which will keep on failing with "context deadline exceeded" while dut is reloading
- when connections are up, we'll see return code unavailable, max allowed timeout is 3 mins
- establish gribi session and do programming